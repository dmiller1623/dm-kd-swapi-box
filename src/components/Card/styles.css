.card {
  display: flex;
  flex-direction: column;
  width: 20rem; }

.card-data {
  color: white; }

.clicked {
  background-color: red; }

.fifth-data {
  font-size: 15px;
  display: flex;
  flex-wrap: wrap;
  padding-bottom: 0 auto; }

h5 {
  margin: 0; }
