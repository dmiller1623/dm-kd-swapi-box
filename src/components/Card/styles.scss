.card {
  display: flex;
  flex-direction: column;
  // flex-wrap: wrap;
  width: 20rem;
}

.card-data {

  color: white
}