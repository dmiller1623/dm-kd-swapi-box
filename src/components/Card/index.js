import React from 'react'
import './styles.css'
import PropTypes from 'prop-types'

const Card = (props) => {
  return(
    <div className='card'>
      <button className={props.evaluateClass(props)} onClick={() => props.updateFavorites(props)}>FAV</button>
      <h2 className='card-data'>{props.first}</h2>
      <h4 className='card-data'>{props.second}</h4>
      <h4 className='card-data'>{props.third}</h4>
      <h4 className='card-data'>{props.fourth}</h4>
      {props.fifth && <div className='fifth-data'>
        <h5 className='card-data'>Residents:</h5>
        {!props.fifth.length && <h5 className='card-data'>0</h5>}
        { 
          props.fifth.map((person, i) => {
            return <h5 key={i} className='card-data'>{person}, </h5>
          })
        }
      </div>}
    </div>
  )
}

Card.propTypes = {
  first: PropTypes.string,
  second: PropTypes.string,
  third: PropTypes.string,
  fourth: PropTypes.string,
  fifth: PropTypes.array
}

export default Card