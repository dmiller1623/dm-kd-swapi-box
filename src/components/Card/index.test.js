import React from 'react'
import Card from './index.js'
import { shallow } from 'enzyme'

describe('Card', () => {
  let wrapper
  let mockFirst
  let mockSecond
  let mockThird
  let mockFourth
  let mockFifth
  let mockEvaluateClass
  let mockUpdateFavorites
  
  beforeEach(() => {
    mockEvaluateClass = jest.fn()
    mockUpdateFavorites = jest.fn()
    
    wrapper = shallow(<Card 
      first='first' 
      second='second'
      third='third'
      fourth='fourth'
      fifth={[{name: 'name'}]}
      evaluateClass={mockEvaluateClass}
      updateFavorites={mockUpdateFavorites}
      />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should invoke updateFavorites and evaluateClass on click', () => {
    wrapper.find('button').simulate('click')

    expect(mockEvaluateClass).toHaveBeenCalled()        
    expect(mockUpdateFavorites).toHaveBeenCalled()
  })

  it('should render a fifth dataset if the data is coming in', () => {
    expect(wrapper.find('.card-data').length).toEqual(6)
    expect(wrapper.find('.fifth-data').length).toEqual(1)

    wrapper = shallow(<Card 
      first='first' 
      second='second'
      third='third'
      fourth='fourth'
      evaluateClass={mockEvaluateClass}
      updateFavorites={mockUpdateFavorites}
      />)

      expect(wrapper.find('.card-data').length).toEqual(4)
      expect(wrapper.find('.fifth-data').length).toEqual(0)

  })
})