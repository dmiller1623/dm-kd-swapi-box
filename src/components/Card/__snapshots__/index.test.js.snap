// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    evaluateClass={[Function]}
    fifth={
        Array [
            Object {
              "name": "name",
            },
          ]
    }
    first="first"
    fourth="fourth"
    second="second"
    third="third"
    updateFavorites={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className={undefined}
          onClick={[Function]}
>
          FAV
</button>,
        <h2
          className="card-data"
>
          first
</h2>,
        <h4
          className="card-data"
>
          second
</h4>,
        <h4
          className="card-data"
>
          third
</h4>,
        <h4
          className="card-data"
>
          fourth
</h4>,
        <div
          className="fifth-data"
>
          <h5
                    className="card-data"
          >
                    Residents:
          </h5>
          <h5
                    className="card-data"
          >
                    Object {
                              "name": "name",
                            }
                    , 
          </h5>
</div>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "FAV",
          "className": undefined,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "FAV",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "first",
          "className": "card-data",
        },
        "ref": null,
        "rendered": "first",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "second",
          "className": "card-data",
        },
        "ref": null,
        "rendered": "second",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "third",
          "className": "card-data",
        },
        "ref": null,
        "rendered": "third",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "fourth",
          "className": "card-data",
        },
        "ref": null,
        "rendered": "fourth",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h5
              className="card-data"
>
              Residents:
</h5>,
            false,
            Array [
              <h5
                className="card-data"
>
                Object {
                                "name": "name",
                              }
                , 
</h5>,
            ],
          ],
          "className": "fifth-data",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Residents:",
              "className": "card-data",
            },
            "ref": null,
            "rendered": "Residents:",
            "type": "h5",
          },
          false,
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                Object {
                  "name": "name",
                },
                ", ",
              ],
              "className": "card-data",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "name": "name",
              },
              ", ",
            ],
            "type": "h5",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className={undefined}
            onClick={[Function]}
>
            FAV
</button>,
          <h2
            className="card-data"
>
            first
</h2>,
          <h4
            className="card-data"
>
            second
</h4>,
          <h4
            className="card-data"
>
            third
</h4>,
          <h4
            className="card-data"
>
            fourth
</h4>,
          <div
            className="fifth-data"
>
            <h5
                        className="card-data"
            >
                        Residents:
            </h5>
            <h5
                        className="card-data"
            >
                        Object {
                                    "name": "name",
                                  }
                        , 
            </h5>
</div>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "FAV",
            "className": undefined,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "FAV",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "first",
            "className": "card-data",
          },
          "ref": null,
          "rendered": "first",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "second",
            "className": "card-data",
          },
          "ref": null,
          "rendered": "second",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "third",
            "className": "card-data",
          },
          "ref": null,
          "rendered": "third",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "fourth",
            "className": "card-data",
          },
          "ref": null,
          "rendered": "fourth",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h5
                className="card-data"
>
                Residents:
</h5>,
              false,
              Array [
                <h5
                  className="card-data"
>
                  Object {
                                    "name": "name",
                                  }
                  , 
</h5>,
              ],
            ],
            "className": "fifth-data",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Residents:",
                "className": "card-data",
              },
              "ref": null,
              "rendered": "Residents:",
              "type": "h5",
            },
            false,
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  Object {
                    "name": "name",
                  },
                  ", ",
                ],
                "className": "card-data",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "name": "name",
                },
                ", ",
              ],
              "type": "h5",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
